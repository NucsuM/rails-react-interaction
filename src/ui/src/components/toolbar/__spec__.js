import React from "react";
import Enzyme, { shallow, mount, configure } from "enzyme";
// import Textbox from "../textbox";
import ToolBar from "./toolbar";
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe("Toolbar", () => {
//   describe("general look", () => {
//     let wrapper;
//     let setApiData = {};

//     beforeEach(() => {
//       wrapper = shallow(<ToolBar setData={setApiData}/>);
//     });

//     it("it have to display all buttons and textboxes", () => {
//       expect(wrapper).toMatchSnapshot();
//   });
// });
  describe("fetch data", () => {
    it("it should filter by a search term (link)", () => {
      // actual test
    });
  });

  describe("clear table", () => {
    it("it clears a filled table", () => {
      // actual test
    });
  });
  describe("submit button", () => {
    it("it should filter by a search term (link)", () => {
      // actual test
    });
  });
});
