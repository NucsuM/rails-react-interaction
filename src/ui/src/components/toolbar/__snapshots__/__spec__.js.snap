// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar general look it have to display all buttons and textboxes 1`] = `
<Fragment>
  <div
    className="grid-container"
  >
    <div
      className="grid-item"
    >
      <button
        className="btn-fetch"
        onClick={[Function]}
        type="button"
      >
        fetch data
      </button>
    </div>
    <div
      className="grid-item"
    >
      <button
        className="btn-clear"
        onClick={[Function]}
        type="button"
      >
        clear table
      </button>
    </div>
    <div
      className="grid-item"
    />
    <div
      className="grid-item"
    />
    <div
      className="grid-item"
    >
      <button
        className="btn-submit"
        onClick={[Function]}
        type="button"
      >
        submit
      </button>
    </div>
    <div
      className="grid-item"
    >
      <Textbox
        class="tb-id"
        name="id"
        onChange={[Function]}
      />
    </div>
    <div
      className="grid-item"
    >
      <Textbox
        class="tb-title"
        name="title"
        onChange={[Function]}
      />
    </div>
    <div
      className="grid-item"
    />
  </div>
</Fragment>
`;
